users/views.py

```diff
@@ -164,4 +171,279 @@ def get(self, request, user_id, token):
+            return self.handle_oauth(provider, access_token)
+        
+        # Fall back to standard email/password auth
+        return super().post(request, *args, **kwargs)
+    
+    def handle_oauth(self, provider, access_token):
+        """
+        Route OAuth authentication requests to the appropriate provider handler.
+        
+        Args:
+            provider: String identifying the OAuth provider ('google' or 'github')
+            access_token: Valid OAuth access token from the provider
+            
+        Returns:
+            Response: JSON response from the provider handler or error if unsupported
+            
+        Raises:
+            HTTP 400: If provider is not supported
+        """
+        if provider == 'google':
+            return self.handle_google_oauth(access_token)
+        elif provider == 'github':
+            return self.handle_github_oauth(access_token)
+        else:
+            return Response(
+                {"error": "Unsupported OAuth provider"}, 
+                status=status.HTTP_400_BAD_REQUEST
+            )
+    
+    def handle_google_oauth(self, access_token):
+        """
+        Authenticate using Google OAuth access token.
+        
+        1. Validates token with Google's API
+        2. Retrieves user profile information
+        3. Creates/updates local user record
+        4. Issues JWT tokens
+        
+        Args:
+            access_token: Valid Google OAuth access token
+            
+        Returns:
+            Response: JSON containing JWT tokens and user data or error message
+            
+        Status Codes:
+            - 200 OK: Successful authentication
+            - 400 Bad Request: Invalid token or missing email
+        """
+        google_userinfo_url = "https://www.googleapis.com/oauth2/v3/userinfo"
+        headers = {"Authorization": f"Bearer {access_token}"}
+
+        try:
+            response = requests.get(google_userinfo_url, headers=headers)
+            response.raise_for_status()
+            data = response.json()
+        except requests.RequestException:
+            return Response(
+                {"error": "Invalid Google token"}, 
+                status=status.HTTP_400_BAD_REQUEST
+            )
+        
+        email = data.get("email")
+        if not email:
+            return Response(
+                {"error": "Email not provided"}, 
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+        first_name = data.get("given_name", "")
+        last_name = data.get("family_name", "")
+
+        user, _ = self.get_or_create_user(
+            email=email,
+            defaults={
+                "first_name": first_name,
+                "last_name": last_name,
+                "password": "",
+                "user_phone": "",
+                "title": "",
+                "role": get_default_role()
+            }
+        )
+
+        return self.generate_jwt_response(user)
```

Comment:
Add a timeout to the requests.get call to Google's API to avoid hanging on slow responses.
Log failed authentication attempts and returned data for debugging.
In error responses, clarify whether the error is due to an invalid token or a network issue.